microsoft/testsuites/network/networksettings.py

```diff
@@ -563,75 +567,117 @@ def validate_device_statistics(self, environment: Environment, log: Logger) -> N
+        final_starved_txq: List[int] = []
+        final_starved_rxq: List[int] = []
+        starved_queues_per_device: Dict[str, Tuple[List[int], List[int]]] = dict()
+        prev_starved_queues_per_device = starved_queues_per_device
 
-        for device_stats in devices_statistics:
-            per_tx_queue_packets: List[int] = []
-            per_rx_queue_packets: List[int] = []
-            nic = client_node.nics.get_nic(device_stats.interface)
-            an_enabled = "AN Disabled "
+        while counter <= 5:
+            self._run_iperf3(server_node, client_node)
 
-            # If AN is enabled on this interface then check the vf stats.
-            if nic.lower:
-                an_enabled = "AN Enabled "
-                try:
-                    device_stats = ethtool.get_device_statistics(nic.lower, True)
-                except UnsupportedOperationException as identifier:
-                    raise SkippedException(identifier)
+            # validate from the stats that no single queue is starved everytime.
+            try:
+                devices_statistics = ethtool.get_all_device_statistics()
+            except UnsupportedOperationException as identifier:
+                raise SkippedException(identifier)
 
-                per_tx_queue_packets = [
-                    v
-                    for (k, v) in device_stats.counters.items()
-                    if self._vf_tx_stats_regex.search(k)
-                ]
+            for device_stats in devices_statistics:
+                per_tx_queue_packets: List[int] = []
+                per_rx_queue_packets: List[int] = []
+
+                nic = client_node.nics.get_nic(device_stats.interface)
+
+                # If AN is enabled on this interface then check the vf stats.
+                if nic.lower:
+                    try:
+                        device_stats = ethtool.get_device_statistics(nic.lower, True)
+                    except UnsupportedOperationException as identifier:
+                        raise SkippedException(identifier)
+
+                    per_tx_queue_packets = [
+                        v
+                        for (k, v) in device_stats.counters.items()
+                        if self._vf_tx_stats_regex.search(k)
+                    ]
+
+                    per_rx_queue_packets = [
+                        v
+                        for (k, v) in device_stats.counters.items()
+                        if self._vf_rx_stats_regex.search(k)
+                    ]
+                else:
+                    per_tx_queue_packets = [
+                        v
+                        for (k, v) in device_stats.counters.items()
+                        if self._tx_queue_stats_regex.search(k)
+                    ]
+
+                    per_rx_queue_packets = [
+                        v
+                        for (k, v) in device_stats.counters.items()
+                        if self._rx_queue_stats_regex.search(k)
```

Comment:
Don't use `self` in test suite. The test suite may be running in multiple test cases together.